### DISCUSSION TITLE
Assumes dependency arguments are query parameters

### DISCUSSION THREAD

**Discussion by chbndrhnns** [fastapi/fastapi#9592] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def test_():
    app = fastapi.FastAPI()

    class Dep:
        def __init__(
            self,
            model_cls: type[pydantic.BaseModel] = None,
        ):
            self._model_cls = model_cls

    @app.get("/get", dependencies=[fastapi.Depends(Dep)])
    async def get():
        return {}

    assert not app.openapi()["paths"]["/get"]["get"].get("parameters")

```

### Description

My expectation is that function arguments only end up in the schema when they use one of the `fastapi.Query`, `fastapi.Header` or `fastapi.Path` defaults.
However, my arbitrary argument `model_cls` also ends up in the schema.

Is there a way to avoid it?

This is the test outcome:

```
>       assert not app.openapi()["paths"]["/get"]["get"].get("parameters")
E       AssertionError: assert not [{'in': 'query', 'name': 'model_cls', 'required': True, 'schema': {'title': 'Model Cls'}}]
```

### Operating System

macOS

### FastAPI Version

0.95.2

### Python Version

3.9.6

**Comment by chrisK824** [comment, marked as answer]
You can make use of custom advanced dependencies like described in the documentation https://fastapi.tiangolo.com/advanced/advanced-dependencies/

Replace concept of `fixed_content` in that example with your arbitrary `model_cls`. Tl;dr story is that you should use an instance of that class in the Depends, and you can declare parameters in __call__ function of the class to assume as arguments/parameters needed in the path

**Reply by chbndrhnns** [reply]
I think FastAPI should only add to the schema what it can identify as relevant for the schema. The fact that it is ok to pass in an instance of model but not the type makes it more plausible to me that this a bug.
